code-experiments/src/c_linear.c

```diff
@@ -217,9 +217,17 @@ static coco_problem_t *c_linear_shuffle(coco_problem_t *problem_c,
   
   /* Pick up a random constraint number other than 1.
    * formula for U[M,N]: 
-   * M + rand() / (RAND_MAX / (N - M + 1) + 1)
+   * random = M + (rand() / (RAND_MAX + 1.0)) * (N - M + 1)
```

Comment:
right, I didn't realize because you apparently used it for a continuous random variable (which I guess means you didn't realize as well ;-) I guess the comment should better mention this as well...